src/luxos/scripts/async_luxos.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+import asyncio
+import json
+
+from .. import misc
+from .. import asyncops
+
+
+async def run(
+    ip_list: list[str], port: int, cmd: str, params: list[str], timeout: float,
+    delay: float | None, details: bool
+) -> None:
+
+    batchsize = 50
+    result = {}
+    for grupid, addresses in enumerate(
+        misc.batched([(ip, port) for ip in ip_list], n=batchsize)
+    ):
+        tasks = []
+        for host, port in addresses:
+            tasks.append(asyncops.execute_command(host, port, timeout, cmd, params, add_address=True))
+        result[grupid] = await asyncio.gather(*tasks, return_exceptions=True)
+
+        # runs only on batchsize, wait delay then proceed onto the next batch
+        if delay:
+            await asyncio.sleep(delay)
+
+    alltasks = [task for group in result.values() for task in group]
+    successes = [task for task in alltasks if not isinstance(task, Exception)]
+    failures = [task for task in alltasks if isinstance(task, Exception)]
+
+    # print a nice report
+    print(f"task executed sucessfully: {len(successes)}")
```

Comment:
We usually use logging instead of prints, this way we can filter on the debug if needed. But no need to refactor everything, I'm putting it as a suggestion.

Reply:
That part of the code is userland, it displays to the user in a nicely formatted way the output. logging shoudl report diagnostic (as opposed to output). We could've used rich, but then that'd be a dependency (luxos.pyz now runs without any external dependency).